src/Microsoft.SqlTools.ServiceLayer/SqlProjects/Contracts/DatabaseReferences/AddSqlProjectReferenceParams.cs

```diff
@@ -0,0 +1,38 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.SqlProjects.Contracts
+{
+    /// <summary>
+    /// Parameters for adding a reference to another SQL project
+    /// </summary>
+    public class AddSqlProjectReferenceParams : AddDatabaseReferenceParams
+    {
+        /// <summary>
+        /// Path to the referenced .sqlproj file
+        /// </summary>
+        public string ProjectPath { get; set; }
+
+        /// <summary>
+        /// GUID for the referenced SQL project
+        /// </summary>
+        public string? ProjectGuid { get; set; }
+
+        /// <summary>
+        /// SQLCMD variable name for specifying the other server this reference is to, if different from that of the current project.
+        /// If this is set, DatabaseVariable must also be set.
+        /// </summary>
+        public string? ServerVariable { get; set; }
+    }
+
+
+    public class AddSqlProjectReferenceRequest
+    {
+        public static readonly RequestType<AddSqlProjectReferenceParams, ResultStatus> Type = RequestType<AddSqlProjectReferenceParams, ResultStatus>.Create("sqlprojects/addSqlProjectReference");
```

Comment:
As far as I am aware, we are only using lower case in the request name in all other places, you might want to be consistent with the current pattern.

Reply:
I was once told they're easier to read if you camel case them. I think @Charles-Gagnon told me that, so I ended up doing it for query execution plan comparison request endpoints..
https://github.com/microsoft/sqltoolsservice/blob/e2962390bc6b521ee33a410611d08384f89cba43/src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/Contracts/ExecutionPlanComparisonRequest.cs#L46

Reply:
It's definitely easier to read IMO, but it doesn't really matter (JSON RPC message names are case insensitive) so whatever is your preference. But readability helps avoid silly spelling mistakes and the like too so there's something to be said for that...

Reply:
@alanrenmsft when choosing from multiple preexisting patterns, I support the use of the one with better readability. :)

Reply:
:) fine by me.